clients/devtools-entry.js

```diff
@@ -9,23 +9,32 @@ const lighthouse = require('../lighthouse-core/index.js');
 const RawProtocol = require('../lighthouse-core/gather/connections/raw.js');
 const log = require('lighthouse-logger');
 const {registerLocaleData, lookupLocale} = require('../lighthouse-core/lib/i18n/i18n.js');
+const desktopDense4G = require('../lighthouse-core/config/constants.js').throttling.desktopDense4G;
 
 /** @typedef {import('../lighthouse-core/gather/connections/connection.js')} Connection */
 
 /**
- * Return a version of the default config, filtered to only run the specified
- * categories. If `lighthouse-plugin-publisher-ads` is in the list of
+ * Returns a config, which runs only certain categories.
+ * Varies the config to use based on device.
+ * If `lighthouse-plugin-publisher-ads` is in the list of
  * `categoryIDs` the plugin will also be run.
  * @param {Array<string>} categoryIDs
+ * @param {string} device
  * @return {LH.Config.Json}
  */
-function getDefaultConfigForCategories(categoryIDs) {
+function createConfig(categoryIDs, device) {
+  /** @type {LH.SharedFlagsSettings} */
+  const settings = {
+    onlyCategories: categoryIDs,
+  };
+  if (device === 'desktop') {
+    settings.throttling = desktopDense4G;
```

Comment:
feels a little weird to manage half the configuration here and half in devtools. should we either bring all of it into here or just export the throttling object? will it be a bigger pain to make changes to devtools if it's all here?

Reply:
> feels a little weird to manage half the configuration here and half in devtools. 

heh we were discussing this point yesterday. :)

but which half in devtools are you talking about?

Reply:
all the `setFlags` business

https://cs.chromium.org/chromium/src/third_party/devtools-frontend/src/front_end/audits/AuditsController.js?type=cs&q=f:audits+-f:out&sq=package:chromium&g=0&l=250-282

Reply:
the flags are set based on values from UI controls in DevTools. So it seems unreasonable to have the entry handle the config entirely.

I was an advocate for just having DevTools pass in an entire config (thus, it would have knowledge of Lighthouse configs). Paul preferred the encapsulation this current approach provides. Although, there's the argument that CDT already knows about the flags, which is a subset of the config ...

Reply:
yeah I'm a little concerned this approach makes it harder for someone to figure out where things are being set, but it's fine